collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -326,4 +334,7 @@ protected void onPostExecute(String result) {
         }
     }
 
+    public boolean checkValidSubmissionURI(String submissionURI) {
```

Comment:
I think what @yanokwa is suggesting is that instead of creating this new method, it would be preferable to just use `UrlUtils.isValidUrl` in the code. It turns out that both do about the same check. Both allow a lot that doesn't actually end up working in this case so we should investigate whether we can improve on that implementation.